src/transactions/types.ts

```diff
@@ -113,13 +114,24 @@ export type AnyRawTransactionInstance = RawTransaction | MultiAgentRawTransactio
  * @group Implementation
  * @category Transactions
  */
-export type InputGenerateTransactionOptions = {
+export type InputGenerateTransactionOptions =
+  | InputGenerateSequenceNumberTransactionOptions
+  | InputGenerateOrderlessTransactionOptions;
+
+export type InputGenerateSequenceNumberTransactionOptions = {
   maxGasAmount?: number;
   gasUnitPrice?: number;
   expireTimestamp?: number;
   accountSequenceNumber?: AnyNumber;
```

Comment:
you can add the `replayNonce: undefined` here and then [here](https://github.com/aptos-labs/aptos-ts-sdk/pull/716/files) simply do `if(options?.replayNonce)`